restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/AccountsMigrationEndpointTest.java

```diff
@@ -0,0 +1,148 @@
+
+@RunWith(Arquillian.class)
+public class AccountsMigrationEndpointTest {
+    final static Logger logger = Logger.getLogger(AccountsMigrationEndpointTest.class);
+    private static final String version = Version.getVersion();
+
+    @ArquillianResource
+    URL deploymentUrl;
+
+    private String adminUsername = "administrator@company.com";
+    private String adminFriendlyName = "Default Administrator Account";
+    private String adminAccountSid = "ACae6e420f425248d6a26948c17a9e2acf";
+    private String adminAuthToken = "77f8c12cc7b8f8423e5c38b035249166";
+    private String adminPassword = "RestComm";
+
+    private String parentUsername = "child@company.com";
+    private String parentSid = "AC574d775522c96f9aacacc5ca60c8c74f";
+    private String parentAuthToken ="77f8c12cc7b8f8423e5c38b035249166";
+
+    @Test
+    public void testMigrateParentAndChildAccount() {
+        String childSid1 = "AC55555555555555555555555555555555";
+
+        String originalOrganizationSid = "ORafbe225ad37541eba518a74248f0ac4c";
+        String newOrganizationSid = "ORafbe225ad37541eba518a74248f0ac4d";
+
+        JsonObject currentParentAccout = RestcommAccountsTool.getInstance().getAccount(deploymentUrl.toString(), adminUsername, adminAuthToken, parentSid);
+        assertNotNull(currentParentAccout);
+        String currentOrg = currentParentAccout.get("organization").getAsString();
+        assertEquals(originalOrganizationSid, currentOrg);
+
+        JsonObject currentChildAccout = RestcommAccountsTool.getInstance().getAccount(deploymentUrl.toString(), adminUsername, adminAuthToken, childSid1);
+        assertNotNull(currentChildAccout);
+        String currentChildOrg = currentChildAccout.get("organization").getAsString();
+        assertEquals(originalOrganizationSid, currentChildOrg);
+
+        //Migrate Parent Account and child accounts
+        RestcommAccountsTool.getInstance().migrateAccount(deploymentUrl.toString(), adminUsername,
+                adminAuthToken, parentSid, newOrganizationSid);
+
+        JsonObject migratedParentAccount = RestcommAccountsTool.getInstance().getAccount(deploymentUrl.toString(), adminUsername, adminAuthToken, parentSid);
+        assertNotNull(migratedParentAccount);
+        String parentNewOrg = migratedParentAccount.getAsJsonObject().get("organization").getAsString();
+        assertEquals(newOrganizationSid, parentNewOrg);
+        assertTrue(!currentOrg.equalsIgnoreCase(parentNewOrg));
+
+        JsonObject childAccount1 = RestcommAccountsTool.getInstance().getAccount(deploymentUrl.toString(), adminUsername, adminAuthToken, childSid1);
+        assertNotNull(childAccount1);
+        String childNewOrg = childAccount1.get("organization").getAsString();
+        assertEquals(newOrganizationSid, childNewOrg);
+
+        //Migrate Parent Account and child accounts
+        assertNotNull(RestcommAccountsTool.getInstance().migrateAccount(deploymentUrl.toString(), adminUsername,
```

Comment:
clarification:what is the purpose of this line? why do we invoke migrate again

Reply:
@jaimecasero if status of the http response on the migration request from `RestcommAccountsTool.getInstance().migrateAccount()` method is 200 OK, the method response will be NOT NULL.
The execution of the migrate method again is needed to move the previously migrated accounts back to the original organization so subsequent test will work as expected